src/System.Collections/src/System/Collections/Generic/SortedList.cs

```diff
@@ -161,11 +161,11 @@ public SortedList(IDictionary<TKey, TValue> dictionary, IComparer<TKey> comparer
                 Debug.Assert(count == _keys.Length);
                 if (count > 1)
                 {
-                    Array.Sort<TKey, TValue>(keys, _values, comparer);
-                    comparer = Comparer; // obtain default if this is null.
+                    // Comparer is set to the default if the comparer parameter is null
+                    Array.Sort<TKey, TValue>(keys, _values, Comparer);
                     for (int i = 1; i != keys.Length; ++i)
                     {
-                        if (comparer.Compare(keys[i - 1], keys[i]) == 0)
+                        if (Comparer.Compare(keys[i - 1], keys[i]) == 0)
```

Comment:
Are you sure about the range check?

I did have a local for `keys[i - 1]` in an earlier revision of what introduced this, but changed to the current form as it's simpler and perhaps clearer as to the intent.

Comparer isn't readonly so this definitely introduces many field reads. I wonder if they would be optimised out if it was readonly. In any case, all we really need here is to move the assignment back to the local `comparer` above the call to Sort, and perhaps expand the comment a bit.